modules/gapi/test/common/gapi_core_tests.hpp

```diff
@@ -30,91 +30,35 @@ enum bitwiseOp
 {
-const char *MathOperations[] = {"ADD", "SUB", "MUL", "DIV"};
-const char *BitwiseOperations[] = {"And", "Or", "Xor"};
-const char *CompareOperations[] = {"CMP_EQ", "CMP_GT", "CMP_GE", "CMP_LT", "CMP_LE", "CMP_NE"};
-//corresponds to OpenCV
-const char *NormOperations[] = {"", "NORM_INF", "NORM_L1", "","NORM_L2"};
-}
-
-
-struct PrintMathOpCoreParams
-{
-    template <class TestParams>
-    std::string operator()(const ::testing::TestParamInfo<TestParams>& info) const
-    {
-        std::stringstream ss;
-        using AllParams = Params<mathOp,bool,double,bool>;
-        const AllParams::params_t& params = info.param;
-        cv::Size sz = AllParams::getCommon<1>(params);  // size
-        ss<<MathOperations[AllParams::getSpecific<0>(params)]  // mathOp
-                    <<"_"<<AllParams::getSpecific<1>(params)  // testWithScalar
-                    <<"_"<<AllParams::getCommon<0>(params)  // type
-                    <<"_"<<(int)AllParams::getSpecific<2>(params)  // scale
-                    <<"_"<<sz.width
-                    <<"x"<<sz.height
-                    <<"_"<<(AllParams::getCommon<2>(params)+1)  // dtype
-                    <<"_"<<AllParams::getCommon<3>(params)  // createOutputMatrices
-                    <<"_"<<AllParams::getSpecific<3>(params);  // doReverseOp
-        return ss.str();
-   }
-};
-
-struct PrintCmpCoreParams
-{
-    template <class TestParams>
-    std::string operator()(const ::testing::TestParamInfo<TestParams>& info) const
+#define CASE(v) case mathOp::v: os << #v; break
+    switch (op)
     {
-        std::stringstream ss;
-        using AllParams = Params<CmpTypes,bool>;
-        const AllParams::params_t& params = info.param;
-        cv::Size sz = AllParams::getCommon<1>(params);  // size
-        ss<<CompareOperations[AllParams::getSpecific<0>(params)]  // CmpType
-                    <<"_"<<AllParams::getSpecific<1>(params)  // testWithScalar
-                    <<"_"<<AllParams::getCommon<0>(params)  // type
-                    <<"_"<<sz.width
-                    <<"x"<<sz.height
-                    <<"_"<<AllParams::getCommon<3>(params);  // createOutputMatrices
-        return ss.str();
-   }
-};
-
-struct PrintBWCoreParams
-{
-    template <class TestParams>
-    std::string operator()(const ::testing::TestParamInfo<TestParams>& info) const
-    {
-        std::stringstream ss;
-        using AllParams = Params<bitwiseOp>;
-        const AllParams::params_t& params = info.param;
-        cv::Size sz = AllParams::getCommon<1>(params);  // size
-        ss<<BitwiseOperations[AllParams::getSpecific<0>(params)]  // bitwiseOp
-                    <<"_"<<AllParams::getCommon<0>(params)  // type
-                    <<"_"<<sz.width
-                    <<"x"<<sz.height
-                    <<"_"<<AllParams::getCommon<3>(params);  // createOutputMatrices
-        return ss.str();
-   }
-};
+        CASE(ADD);
+        CASE(SUB);
+        CASE(MUL);
+        CASE(DIV);
```

Comment:
missing default?

Reply:
hmm, nothing bad should happen AFAIK. but nothing good either (except the warning). I agree with you, seems wise to add `default: throw () ...` just so that it's easier to see if some value is missing. will update

Reply:
Let it be CV_Assert.

Reply:
I think CV_Assert is more about internal OpenCV error-reporting (am I wrong?). Added `throw runtime_error()` for now, can redo with CV_Assert if required

Reply:
CV_Assert does exactly the same. Just use CV_Assert

Reply:
Note we also have our own GAPI_Assert (which used to be more portable when we supported LeonOS directly).
```
GAPI_Assert(false && "your text");
```

Reply:
@dmatveev done with GAPI_Assert